system-addon/lib/HighlightsFeed.jsm

```diff
@@ -41,13 +41,26 @@ this.HighlightsFeed = class HighlightsFeed {
 
   postInit() {
     SectionsManager.enableSection(SECTION_ID);
+    this.fetchHighlights(true);
   }
 
   uninit() {
     SectionsManager.disableSection(SECTION_ID);
   }
 
   async fetchHighlights(broadcast = false) {
+    // We need TopSites to have been initialised for deduping
+    if (!this.store.getState().TopSites.initialized) {
+      await new Promise(resolve => {
+        const unsubscribe = this.store.subscribe(async () => {
```

Comment:
is this `async` required since there are no `await`s inside? I am still an async/await newb so I'm not sure.

Reply:
Ah no it's not that's a good point, cheers